src/services/aave/subgraph/entities/reserves/index.ts

```diff
@@ -0,0 +1,79 @@
+import { ComputedReserveData, v2 } from '@aave/protocol-js';
+
+import { TokenPrices } from '@/services/coingecko/api/price.service';
+import { Pool } from '@/services/balancer/subgraph/types';
+
+import { FiatCurrency } from '@/constants/currency';
+
+import { bnum } from '@/lib/utils';
+
+import Service from '../../aave-subgraph.service';
+
+import queryBuilder from './query';
+
+import { QueryBuilder } from '@/types';
+
+export default class Reserves {
+  service: Service;
+  query: QueryBuilder;
+
+  constructor(service: Service, query: QueryBuilder = queryBuilder) {
+    this.service = service;
+    this.query = query;
+  }
+
+  public async get(args = {}, attrs = {}): Promise<ComputedReserveData[]> {
+    const query = this.query(args, attrs);
+    const { reserves } = await this.service.client.get(query);
+
+    return v2.formatReserves(reserves);
+  }
+
+  public async calcWeightedSupplyAPRFor(
```

Comment:
This function seems like it shouldn't be in the subgraph service. It looks like it's a more general Aave service function, whereas this file is specifically about interacting with the reserves entity or formatting the responses. I would move it to a `/services/aave/aave.service.ts` class that has the subgraph service as an injected dependency.

Reply:
Good idea!